qubes/ext/pci.py

```diff
@@ -227,6 +228,9 @@ def on_device_pre_attached_pci(self, vm, event, device):
             return
 
         try:
+            device = next(
+                self.on_device_get_pci(vm, event, device.ident)
+            )
```

Comment:
I wonder whether more readable would be to put `PCIDevice(vm, device.ident)` here directly... And since this is repeated multiple times below, definitely it will be less code (in characters ;) ) to duplicate.